### PR TITLE
refactor: ziofy ValuesRouteV1 (DEV-2012)

### PR DESCRIPTION
**Pull request by seakayone**
### Task Description/Number

Issue Number: DEV-2012

### Basic Requirements

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### PR Type

What kind of change does this PR introduce?

- [ ] fix: represents bug fixes
- [x] refactor: represents production code refactoring
- [ ] feat: represents a new feature
- [ ] docs: documentation changes (no production code change)
- [ ] chore: maintenance tasks (no production code change)
- [ ] test: all about tests: adding, refactoring tests (no production code change)
- [ ] other... Please describe:

### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No
- [ ] Maybe (not 100% sure => check with FE)

### Does this PR change client-test-data?

- [ ] Yes (don't forget to update the JS-LIB team about the change)
- [x] No

### CODE REVIEW

File: webapi/src/main/scala/org/knora/webapi/messages/StringFormatter.scala [hunk 1]
```diff
@@ -577,7 +578,7 @@ class StringFormatter private (
   initForTest: Boolean = false
 ) {
 
-  import StringFormatter._
+  import org.knora.webapi.messages.StringFormatter._
```
Comment (irinaschubert) [inline webapi/src/main/scala/org/knora/webapi/messages/StringFormatter.scala:581]: Could that be moved to the beginning of the file as Codacy reports?

File: webapi/src/main/scala/org/knora/webapi/routing/RouteUtilV1.scala [hunk 2]
```diff
@@ -127,17 +130,14 @@ object RouteUtilV1 {
...
-        textWithStandoffTagV1 <-
-          ZIO.attempt(
-            StandoffTagUtilV2.convertXMLtoStandoffTagV2(xml, mappingResponse, acceptStandoffLinksToClientIDs, log)
-          )
-      } yield textWithStandoffTagV1
-    }
+  ): ZIO[MessageRelay, Throwable, TextWithStandoffTagsV2] =
+    for {
+      mappingResponse <- MessageRelay.ask[GetMappingResponseV2](GetMappingRequestV2(mappingIri, userProfile))
+      textWithStandoffTagV1 <-
```
Comment (irinaschubert) [inline webapi/src/main/scala/org/knora/webapi/routing/RouteUtilV1.scala:136]:
```suggestion
      textWithStandoffTagV2 <-
```

Comment (irinaschubert) [reply webapi/src/main/scala/org/knora/webapi/routing/RouteUtilV1.scala:136]: Not sure if V1 is on purpose or not.. (not that it matters a lot as it is only a variable name)

File: webapi/src/main/scala/org/knora/webapi/routing/RouteUtilV1.scala [hunk 3]
```diff
@@ -273,4 +273,34 @@ object RouteUtilV1 {
     } else {
       throw BadRequestException(s"MIME type ${fileMetadataResponse.internalMimeType} not supported in Knora API v1")
     }
+
+  def verifyNumberOfParams[A](
```
Comment (irinaschubert) [inline webapi/src/main/scala/org/knora/webapi/routing/RouteUtilV1.scala:277]: Where are this and the following functions coming from..?

**Review by irinaschubert — APPROVED** [review]
I couldn't properly review the changes in `ValuesRouteV1` as there were that many changes. But I guess it is OK. There are a lot of Codacy complaints about missing tests and bad code (e.g. get() on an option) but I guess it's out of scope here..

File: webapi/src/main/scala/org/knora/webapi/routing/RouteUtilV1.scala (hunk 3 above)
Comment (seakayone) [reply webapi/src/main/scala/org/knora/webapi/routing/RouteUtilV1.scala:278]: I have extracted duplicate code in the Route itself in order to make it easily reusable within the other routes.

File: webapi/src/main/scala/org/knora/webapi/messages/StringFormatter.scala (hunk 1 above)
Comment (seakayone) [reply webapi/src/main/scala/org/knora/webapi/messages/StringFormatter.scala]: done during the merge with main.

File: webapi/src/main/scala/org/knora/webapi/routing/RouteUtilV1.scala [hunk 4]
```diff
@@ -273,4 +274,32 @@ object RouteUtilV1 {
...
+    params: Seq[A],
+    errorMessage: String,
+    length: Int
+  ): IO[BadRequestException, Option[Nothing]] =
+    ZIO.fail(BadRequestException(errorMessage)).when(params.length != length)
+
+  def toSparqlEncodedString(str: String, errorMsg: String): ZIO[StringFormatter, BadRequestException, IRI] =
```
Comment (mpro7) [inline webapi/src/main/scala/org/knora/webapi/routing/RouteUtilV1.scala:285]: If this is here defined, is the function named the same in SF ` def toSparqlEncodedString(s: String): Option[String]` used at all?

**Review by mpro7 — APPROVED** [review]
It's really hard to read diff of `ValuesRouteV1.scala`, but at the end the outcome looks way cleaner. Thanks! Below few remarks:

**Comment by seakayone** [comment]
> I couldn't properly review the changes in `ValuesRouteV1` as there were that many changes. But I guess it is OK. There are a lot of Codacy complaints about missing tests and bad code (e.g. get() on an option) but I guess it's out of scope here..

The complaints about the `.get` on an Option value were present before. Probably we have to fail with a `BadRequest` in these cases as well. In those cases I guess the assumption is that the value must be filled as part of that particular request. Before the user would see a 500 in those cases. It would be an improvement if this 

wdyt, @irinaschubert ?

**Comment by irinaschubert** [comment]
> > I couldn't properly review the changes in `ValuesRouteV1` as there were that many changes. But I guess it is OK. There are a lot of Codacy complaints about missing tests and bad code (e.g. get() on an option) but I guess it's out of scope here..
> 
> The complaints about the `.get` on an Option value were present before. Probably we have to fail with a `BadRequest` in these cases as well. In those cases I guess the assumption is that the value must be filled as part of that particular request. Before the user would see a 500 in those cases. wdyt, @irinaschubert ?

Yes, a Bad Request sounds reasonable in that case.

**Comment by seakayone** [comment]
> > > I couldn't properly review the changes in `ValuesRouteV1` as there were that many changes. But I guess it is OK. There are a lot of Codacy complaints about missing tests and bad code (e.g. get() on an option) but I guess it's out of scope here..
> > 
> > 
> > The complaints about the `.get` on an Option value were present before. Probably we have to fail with a `BadRequest` in these cases as well. In those cases I guess the assumption is that the value must be filled as part of that particular request. Before the user would see a 500 in those cases. wdyt, @irinaschubert ?
> 
> Yes, a Bad Request sounds reasonable in that case.

see PR https://github.com/dasch-swiss/dsp-api/pull/2561

**Pull request merged**